src/main/java/com/cryptomorin/xseries/skull/XSkull.java

```diff
@@ -480,19 +688,33 @@ private static String extractTextureHash(String input) {
         else throw new IllegalArgumentException("Invalid input: " + input);
     }
 
+    /**
+     * Checks if the provided {@link GameProfile} has a texture property.
+     *
+     * @param profile The {@link GameProfile} to check.
+     * @return {@code true} if the profile has a texture property, {@code false} otherwise.
+     */
     private static boolean hasTextures(GameProfile profile) {
         return Iterables.getFirst(profile.getProperties().get("textures"), null) != null;
     }
 
-    private static String encodeBase64(@Nonnull String str) {
+    /**
+     * Encodes the provided string into Base64 format.
+     *
+     * @param str The string to encode.
+     * @return The Base64 encoded string.
+     */
+    private static String encodeBase64(String str) {
         return Base64.getEncoder().encodeToString(str.getBytes(StandardCharsets.UTF_8));
     }
 
     /**
      * Tries to decode the string as a Base64 value.
-     * @return the decoded Base64 string if it is a Base64 string.
+     *
+     * @param base64 The Base64 string to decode.
+     * @return An {@link Optional} containing the decoded Base64 string if it is a valid Base64 string, or empty if not.
      */
-    private static Optional<String> decodeBase64(@Nonnull String base64) {
+    private static Optional<String> decodeBase64(String base64) {
```

Comment:
Same as the previous code, specially since this is an internal code, it shouldn't use optionals. Just return null.